app/src/main/kotlin/com/wire/android/ui/WireActivity.kt

```diff
@@ -277,22 +278,33 @@ class WireActivity : AppCompatActivity() {
     @Composable
     private fun joinConversationDialog(joinedDialogState: JoinConversationViaCodeState?, navigate: (NavigationCommand) -> Unit) {
         joinedDialogState?.let {
+
+            val onComplete: (convId: ConversationId?) -> Unit = remember {
+                {
+                    it?.also {
+                        appLogger.d("Join conversation via code dialog completed, navigating to conversation screen")
+                        navigate(NavigationCommand(ConversationScreenDestination(it), BackStackMode.CLEAR_TILL_START))
+                        viewModel.onJoinConversationFlowCompleted()
+                    }
+                }
+            }
+
             when (it) {
                 is JoinConversationViaCodeState.Error -> JoinConversationViaInviteLinkError(
                     errorState = it,
-                    onCancel = viewModel::cancelJoinConversation
+                    onCancel = { onComplete(null) }
```

Comment:
onComplete does one of 2 things, if there is no conversation is provided it will just close the dialog, otherwise it will navigate to that conversation